pkg/controller/deployment/deployment_controller.go

```diff
@@ -509,6 +510,15 @@ func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Dep
 		return dc.updateDeploymentStatus(allRSs, newRS, deployment)
 	}
 
+	// If we need to create a new RS, create it now
+	if newRS == nil {
+		newRS, oldRSs, err = dc.getNewAndOldReplicaSets(deployment, true)
```

Comment:
This means we are going to relist all rcs just to create the new. Can you add a TODO to fix this?